src/codegen/mod.rs

```diff
@@ -2316,9 +2263,37 @@ impl CodeGenerator for ObjCInterface {
         let mut trait_items = vec![];
 
         for method in self.methods() {
-            let method_name = ctx.rust_ident(method.name());
+            let signature = method.signature();
+            let fn_args = utils::fnsig_arguments(ctx, signature);
+            let fn_ret = utils::fnsig_return_ty(ctx, signature);
+            let sig = aster::AstBuilder::new()
+                .method_sig()
+                .unsafe_()
+                .fn_decl()
+                .self_()
+                .build(ast::SelfKind::Value(ast::Mutability::Immutable))
+                .with_args(fn_args.clone())
+                .build(fn_ret);
+
+            // Collect the actual used argument names
+            let arg_names: Vec<_> = fn_args.iter()
+                .map(|ref arg| {
+                    let pat = arg.pat.clone().unwrap();
+                    match pat.node {
+                        ast::PatKind::Ident(_, spanning, _) => {
```

Comment:
not super awesome, but I guess it works. Alternatively, maybe get an `Option<&mut Vec<String>>` in `fnsig_arguments` that gets the argument names? That could also get used by `arguments_for_signature` in `src/codegen/helpers.rs`.